src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,175 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    public static class Node<T> {
+        private Node<T> prevNode;
+        private Node<T> nextNode;
+        private T value;
+
+        public Node(Node<T> prevNode, T value, Node<T> nextNode) {
+            this.prevNode = prevNode;
+            this.value = value;
+            this.nextNode = nextNode;
+        }
+
+        public Node(T value) {
+            this.value = value;
+        }
+    }
+
+    private int size;
+    private Node<T> head = null;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = newNode;
+            tail = newNode;
+            size++;
+            return;
+        }
+        tail.nextNode = newNode;
+        newNode.prevNode = tail;
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        validateIndexAdd(index);
+        if (index == size) {
+            add(value);
```

Comment:
if we extract code in add() method like in previous comment, here would be call of linkTail method